test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -551,6 +566,232 @@ public void narrativeDecoration() throws Exception {
+            throw e;
+        }
+
+        assertThat("incorrect search objects count", noAuthSearchObjsOut.getObjects().size(), is(1));
+        TestCommon.compare(noAuthSearchObjsOut.getObjects().get(0), expected2);
+
+        assertThat("incorrect search objects count", authSearchObjsOut.getObjects().size(), is(2));
+
+        Set<String> res = new HashSet<>();
+        res.add(expected1.getGuid());
+        res.add(expected2.getGuid());
+        assertThat("incorrect results", res.contains(authSearchObjsOut.getObjects().get(0).getGuid()), is(true));
+        assertThat("incorrect results", res.contains(authSearchObjsOut.getObjects().get(1).getGuid()), is(true));
+
+    }
+    @Test
+    public void optionalAuthForSearchTypes() throws Exception {
+        setUpOptionalAuthForSearch();
+
+        SearchTypesOutput authSearchTypesOut;
+        SearchTypesOutput noAuthSearchTypesOut;
+
+        try {
+            noAuthSearchTypesOut = noAuthSearchCli.searchTypes(new SearchTypesInput()
+                    .withMatchFilter(new MatchFilter())
+                    .withAccessFilter(new AccessFilter()));
+
+            authSearchTypesOut = searchCli.searchTypes(new SearchTypesInput()
+                    .withMatchFilter(new MatchFilter())
+                    .withAccessFilter(new AccessFilter()));
+
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        Map<String, Long> typeToCount = noAuthSearchTypesOut.getTypeToCount();
+        assertThat("incorrect search types map size", typeToCount.size(), is(1));
+        assertThat("Incorrect search types type name", typeToCount.keySet(), is(set("Deco")));
+        assertThat("Incorrect search types type count", typeToCount.get("Deco"), is(Long.valueOf(1)));
+
+        typeToCount = authSearchTypesOut.getTypeToCount();
+        assertThat("incorrect search types map size", typeToCount.size(), is(1));
+        assertThat("Incorrect search types type name", typeToCount.keySet(), is(set("Deco")));
+        assertThat("Incorrect search types type count", typeToCount.get("Deco"), is(Long.valueOf(2)));
+    }
+
+    @Test
+    public void optionalAuthForGetObjects() throws Exception {
+        setUpOptionalAuthForSearch();
+
+        GetObjectsOutput authGetObjsOut;
+        GetObjectsOutput noAuthGetObjsOut;
+
+        try {
+            noAuthGetObjsOut = noAuthSearchCli.getObjects(new GetObjectsInput()
+                    .withGuids(Arrays.asList("WS:1/1/1", "WS:2/1/1")));
+
+            authGetObjsOut = searchCli.getObjects(new GetObjectsInput()
+                    .withGuids(Arrays.asList("WS:1/1/1", "WS:2/1/1")));
+
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        assertThat("incorrect get objects count", noAuthGetObjsOut.getObjects().size(), is(0));
```

Comment:
Since one of the objects is public shouldn't there be one object returned? Seems like a bug...

Reply:
so getObjects lets you get objects you have access to (it checks if you have the right access group id and doesnt care about public/private). Is that a feature or bug?

Reply:
Also it unclear to me whats the usage of GetObjects if there is SearchObjects

Reply:
For get objects there's no way to specify private/public access, so you're right. The purpose of get objects is to allow getting objects directly by the guid. I assume it's a special case of search objects. It's weird in that it allows for a match filter *and* a guid list.

Reply:
Actually looking at the code, the get_objects match filter is ignored. Nice. That should be taken out of the spec.

Reply:
`    public GetObjectsOutput getObjects(final GetObjectsInput params, final String user)
            throws Exception {
        final Set<Integer> accessGroupIDs;
        final long t1 = System.currentTimeMillis();
        if (user != null) {
            accessGroupIDs = new HashSet<>(accessGroupProvider.findAccessGroupIds(user));
        } else {
            accessGroupIDs = Collections.emptySet();
        }
        final Set<GUID> guids = new LinkedHashSet<>();
        for (final String guid : params.getGuids()) {
            final GUID g = new GUID(guid);

            //TODO DP this is a quick fix for now, doesn't take data palettes into account
            if (accessGroupIDs.contains(g.getAccessGroupId())) {
                // don't throw an error, just don't return data
                guids.add(g);
            }
        }

        final kbasesearchengine.search.PostProcessing postProcessing =
                toSearch(params.getPostProcessing());
        final List<kbasesearchengine.search.ObjectData> objs = indexingStorage.getObjectsByIds(
                guids, postProcessing);
        final GetObjectsOutput ret = new GetObjectsOutput().withObjects(objs.stream()
                .map(this::fromSearch).collect(Collectors.toList()));
        ret.withSearchTime(System.currentTimeMillis() - t1);
        return ret;
    }`

I am looking at the getObjects in SearchMethods. It  looks like it was intentionally made that an unauth user can't get objects regardless of its public since its filtering the guids before calling elasticsearch. where is the get_objects match filter??

Reply:
Right, because before this PR all users were authed. Now unauthed users are allowed, that behavior is going to cause unauthed users to never get data, IIUC.

> where is the get_objects match filter??

https://github.com/kbase/KBaseSearchEngine/blob/master/KBaseSearchEngine.spec#L303

Don't worry about that for now though. Take it out in another PR.